pulp_smash/tests/puppet/api_v2/test_sync_publish.py

```diff
@@ -0,0 +1,650 @@
+        server_config.base_url + href + 'search/units/',
+        json={'criteria': {}},
+        **server_config.get_requests_kwargs()
+    ), responses)
+
+
+def _query_repo(server_config, query_url, repo_id, responses=None):
+    """Query pulp puppet repo for presence of given unit.
+
+    :param server_config: A :class:`pulp_smash.config.ServerConfig` object.
+    :param query_url: Url in Puppet Forge formatted URL.
+    :param repo_id: id of queried repository
+    :param responses: A list, or some other object that supports the ``append``
+        method. If given, all server responses are appended to this object.
+    :returns: The server's JSON-decoded response.
+
+    """
+    kwargs = server_config.get_requests_kwargs()
+    kwargs['auth'] = ('repository', repo_id)
+    return handle_response(requests.get(
+        server_config.base_url + query_url,
+        **kwargs
+    ), responses)
+
+
+def _start_content_upload(server_config, responses=None):
+    """Start a content upload.
+
+    :param server_config: A :class:`pulp_smash.config.ServerConfig` object.
+    :param responses: A list, or some other object that supports the ``append``
+        method. If given, all server responses are appended to this object.
+    :returns: The server's JSON-decoded response.
+
+    """
+    return handle_response(requests.post(
+        server_config.base_url + _CONTENT_UPLOADS_PATH,
+        **server_config.get_requests_kwargs()
+    ), responses)
+
+
+def _upload_file(server_config, href, content, responses=None):
+    """Upload a file and return the server's response.
+
+    :param server_config: A :class:`pulp_smash.config.ServerConfig` object.
+    :param href: A string. The path to an open upload request.
+    :param content: The content of the file to upload.
+    :param responses: A list, or some other object that supports the ``append``
+        method. If given, all server responses are appended to this object.
+    :returns: The server's JSON-decoded response.
+
+    """
+    return handle_response(requests.put(
+        server_config.base_url + href + '0/',
+        data=content,
+        **server_config.get_requests_kwargs()
+    ), responses)
+
+
+def _import_puppet_to_repo(server_config, upload_id, href, responses=None):
+    """Import an puppet from an upload into a repository.
+
+    :param server_config: A :class:`pulp_smash.config.ServerConfig` object.
+    :param upload_id: A string. The ID of an upload request.
+    :param href: A string. The path to an puppet repository.
+    :param responses: A list, or some other object that supports the ``append``
+        method. If given, all server responses are appended to this object.
+    :returns: The server's JSON-decoded response.
+
+    """
+    return handle_response(requests.post(
+        server_config.base_url + href + 'actions/import_upload/',
+        json={'unit_key': {}, 'unit_type_id': 'puppet_module', 'upload_id': upload_id},  # noqa pylint:disable=line-too-long
```

Comment:
Please wrap these lines. There's no compelling reason to exceed a line length limit here.